sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableServiceAsyncClientTest.java

```diff
@@ -419,25 +427,87 @@ public void canUseSasTokenToCreateValidTableClient() {
+
+        List<TableServiceCorsRule> corsRules = new ArrayList<>();
+        corsRules.add(new TableServiceCorsRule()
+            .setAllowedMethods("GET,PUT,HEAD")
+            .setAllowedOrigins("*")
+            .setAllowedHeaders("x-ms-version")
+            .setExposedHeaders("x-ms-client-request-id")
+            .setMaxAgeInSeconds(10));
+
+        TableServiceMetrics hourMetrics = new TableServiceMetrics()
+            .setEnabled(true)
+            .setVersion("1.0")
+            .setRetentionPolicy(retentionPolicy)
+            .setIncludeApis(true);
+
+        TableServiceMetrics minuteMetrics = new TableServiceMetrics()
+            .setEnabled(true)
+            .setVersion("1.0")
+            .setRetentionPolicy(retentionPolicy)
+            .setIncludeApis(true);
+
+        TableServiceProperties properties = new TableServiceProperties()
+            .setLogging(logging)
+            .setCorsRules(corsRules)
+            .setMinuteMetrics(minuteMetrics)
+            .setHourMetrics(hourMetrics);
+
+        StepVerifier.create(serviceClient.setProperties(properties))
+            .expectComplete()
+            .verify();
+
+        StepVerifier.create(serviceClient.getProperties())
+            .assertNext(retrievedProperties -> {
+                assertNotNull(retrievedProperties);
+                assertNotNull(retrievedProperties.getCorsRules());
+                assertEquals(1, retrievedProperties.getCorsRules().size());
+                assertNotNull(retrievedProperties.getCorsRules().get(0));
+                assertNotNull(retrievedProperties.getHourMetrics());
+                assertNotNull(retrievedProperties.getMinuteMetrics());
+                assertNotNull(retrievedProperties.getLogging());
+            })
+            .expectComplete()
+            .verify();
     }
 
     @Test
-    public void getStatistics() {
-        TableServiceStatistics statistics =  serviceClient.getStatistics().block();
+    public void getStatistics() throws URISyntaxException {
+        URI primaryEndpoint = new URI(serviceClient.getServiceEndpoint());
+        String[] hostParts = primaryEndpoint.getHost().split("\\.");
+        StringJoiner secondaryHostJoiner = new StringJoiner(".");
+        secondaryHostJoiner.add(hostParts[0] + "-secondary");
+
+        for (int i = 1; i < hostParts.length; i++) {
+            secondaryHostJoiner.add(hostParts[i]);
+        }
 
-        assertNotNull(statistics);
-        assertNotNull(statistics.getGeoReplication());
-        assertNotNull(statistics.getGeoReplication().getStatus());
-        assertNotNull(statistics.getGeoReplication().getLastSyncTime());
+        String secondaryEndpoint = primaryEndpoint.getScheme() + "://" + secondaryHostJoiner;
+
+        TableServiceAsyncClient secondaryClient = new TableServiceClientBuilder()
+            .endpoint(secondaryEndpoint)
+            .serviceVersion(serviceClient.getServiceVersion())
+            .pipeline(serviceClient.getHttpPipeline())
```

Comment:
A new pipeline is created if it's not provided in the builder. So, this isn't required.

Reply:
I thought it would be better to use the same pipeline from the first client to use the same configuration. I think can write things better to make it so that we can get those settings in the test class itself.